bciers/apps/reporting/src/app/components/activities/ActivityForm.tsx

```diff
@@ -156,7 +156,7 @@ export default function ActivityForm({
     const sourceTypeCount = Object.keys(sourceTypeMap).length;
     // Ensure we use the filtered formData with omitted extra data
     const filteredData = e.formData;
-    const selectedSourceTypeData = Object.keys(filteredData.sourceTypes);
+    const selectedSourceTypeData = Object.keys(filteredData.sourceTypes ?? {});
```

Comment:
This would let a user submit no data for an activity they said they would report for. Is this intended?